### ISSUE TITLE
About data augmentations in CityScapes

### ISSUE THREAD

**Issue by mmm-cc** [EnVision-Research/MTMamba#8]
Thank you for your excellent work! I have some questions regarding the data augmentation details for the CityScapes dataset in MTMABA++.

In NYUD and Pascal-Context, the previous methods typically set a template size (NY: [448, 576], PS: [512, 512]) that is larger than all the dataset image sizes. This allows for simple padding of the original images during the testing phase, without worrying about changes in label values caused by reshaping operations. Similarly, in TaskPrompter, a larger size of [1024, 2048] is used, which exceeds the original image dimensions.

However, I noticed that in MTMAMA++, you have set the template size to [512, 1024], which is smaller and may lead to label value discrepancies. Could you provide more details on this choice?

**Comment by Baijiong-Lin** [comment]
In all datasets, the labels are resized to the same size as the input images. We use a smaller size in CityScapes due to the limited computational resources.

**Comment by mmm-cc** [comment]
Thank you very much!

**Comment by mmm-cc** [comment]
What are the loss function and weight settings for CityScapes? I replicated MTMamba using the same loss function and weights as in Taskprompter (100 for semantic segmentation and 1 for depth), but the results did not meet the expectations. Can you give some details?

![Image](https://github.com/user-attachments/assets/f0547984-82c8-4af9-a5f6-eee8a06ee020)

![Image](https://github.com/user-attachments/assets/b788d540-96db-4ddd-ac82-c45dbc228271)

**Comment by Baijiong-Lin** [comment]
The log loss weighting is used in Cityscapes.

**Comment by mmm-cc** [comment]
Thank you for your patient responses！

**Comment by mmm-cc** [comment]
I employed log loss on the Cityscapes dataset, assigning a weight of 1 for both semantic segmentation and depth. The input and labels were resized to [512, 1024]. The results were comparable to those obtained without loss weighting (as shown in the above figure). Could you provide the loss weights and additional details?

**Comment by Baijiong-Lin** [comment]
The loss weights of both semantic segmentation and depth estimation are set as 1.

**Comment by mmm-cc** [comment]
Thank you for your response. I may have made some errors in the details, and I will check again.

**Comment by Baijiong-Lin** [comment]
I have updated the code for the Cityscapes experiment. You can check and try it. BTW, in the Cityscapes experiment, we resize input images to [512, 1024] and interpolate predictions to [1024, 2048]. Hence, we do not resize the labels. Sorry for the wrong details I provided before.

**Comment by mmm-cc** [comment]
> I have updated the code for the Cityscapes experiment. You can check and try it. BTW, in the Cityscapes experiment, we resize input images to [512, 1024] and interpolate predictions to [1024, 2048]. Hence, we do not resize the labels. Sorry for the wrong details I provided before.

Thanks a lot, I will try again.

**Comment by mmm-cc** [comment]
![Image](https://github.com/user-attachments/assets/aa33b987-b108-4c54-85c5-e7adf424a1b2)
Thank you for updating the code. I still have some questions and would appreciate your help. I downloaded the latest version of the code and only modified the datapath in ./configs/mypath.py, then tested MTMamba on the Cityscapes dataset using the pre-trained parameters you provided. The semseg task results are consistent with the article, but there are still issues with the depth task. Can you check if the pre-trained model is correct or if I made any mistakes in the details?

**Comment by Baijiong-Lin** [comment]
![Image](https://github.com/user-attachments/assets/54470fe8-0abf-4253-9b1a-ac335f209a9b)

I have checked the pre-trained checkpoint of MTMamba. It is correct.

**Comment by Baijiong-Lin** [comment]
Did you modify the testing batch size? I note you only have 84 batches. Cityscapes has 500 validation samples and we set the testing batch size as 4, thus it should be 125 batches.

**Comment by Baijiong-Lin** [comment]
I set the testing batch size as 6, thus we have 84 batches. The results are still consistent with our paper.

 
![Image](https://github.com/user-attachments/assets/fe18b5b8-3633-4eb7-9e2b-9b66c0d02c5f)

**Comment by mmm-cc** [comment]
![Image](https://github.com/user-attachments/assets/9c3a7307-0233-4bf9-acb9-a40aebd77a36)
Thank you very much for your reminder. I have successfully reproduced the performance. I wrote a script to perform model inference, but mistakenly switched to my own data processing folder in the script. Once again, thank you for your detailed response. Best wishes to you!
